ntrans_gui.py

```diff
@@ -143,67 +141,69 @@ def __init__(self) -> None:
 
-        for checkbox in n_to_ngram_checkbox.values():
+        for checkbox in self.n_to_ngram_checkbox.values():
             checkbox.grid(sticky="W", column=0, padx=(40, 0), pady=(0, 0))
 
-        data_size_label.grid(sticky="W", column=0, padx=(20, 0), pady=(30, 0))
-        ngram_data_size.grid(sticky="W", column=0, padx=(20, 0), pady=(5, 0))
+        self.data_size_label.grid(sticky="W", column=0, padx=(20, 0), pady=(30, 0))
+        self.ngram_data_size.grid(sticky="W", column=0, padx=(20, 0), pady=(5, 0))
 
-        target_language_label.grid(sticky="W", column=0, padx=(20, 20), pady=(30, 0))
-        target_language.grid(sticky="W", column=0, padx=(20, 20), pady=(0, 0))
+        self.target_language_label.grid(sticky="W", column=0, padx=(20, 20), pady=(30, 0))
+        self.target_language.grid(sticky="W", column=0, padx=(20, 20), pady=(0, 0))
 
-        filepath_label.grid(sticky="W", column=0, padx=(20, 0), pady=(30, 10))
-        show_filepath.grid(sticky="W", column=0, padx=(20, 0), pady=(0, 0))
-        get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 0))
+        self.filepath_label.grid(sticky="W", column=0, padx=(20, 0), pady=(30, 10))
+        self.show_filepath.grid(sticky="W", column=0, padx=(20, 0), pady=(0, 0))
+        self.get_savepath_button.grid(sticky="W", column=1, row=13, padx=(0, 20), pady=(0, 0))
 
-        generate_dictionary.grid(column=0, columnspan=2, padx=(0, 0), pady=(0, 0))
+        self.generate_dictionary.grid(column=0, columnspan=2, padx=(0, 0), pady=(0, 0))
         self.setting_not_defined_warning.grid(column=0, row=15, columnspan=2, padx=(0, 0), pady=(0, 0))
         self.estimated_time_label.grid(sticky="W", column=0, columnspan=2, padx=(30, 0), pady=(0, 0))
 
-        about_help_buttonframe.grid(sticky="E", column=0, row=17, columnspan=2, padx=(20, 20), pady=(20, 10))
-
-        # Black turn on formatting
+        self.about_help_buttonframe.grid(sticky="E", column=0, row=17, columnspan=2, padx=(20, 20), pady=(20, 10))
         # fmt: on
 
-        self.filepath.trace_add("write", self.update_user_choice_vars)
-        self.target_language_var.trace_add("write", self.update_user_choice_vars)
-        for var in self.checkbox_vars.values():
-            var.trace_add("write", self.update_user_choice_vars)
-            var.trace_add("write", self.update_estimated_time_label)
-
-        self.data_size_var.trace_add("write", self.update_user_choice_vars)
-
-        self.select_all_var.trace_add("write", self.update_estimated_time_label)
-        self.data_size_var.trace_add("write", self.update_estimated_time_label)
-
-        self.update_estimated_time_label()
+    def get_target_languages(self) -> List[str]:
+        with open("target_languages.json", "r") as target_language_file:
+            target_languages = json.load(target_language_file)
+            return target_languages["target_languages"]  # type: ignore
```

Comment:
You don't need `type: ignore` if you use a local variable with a type hint. The advantage is that because we are not blindly ignoring all errors, mypy would complain if it knew that `json.load()` gives something that can't possibly be `List[str]`.

```suggestion
            result: List[str] = json.load(target_language_file)["target_languages"]
            return result
```